### PR TITLE
python312Packages.bugzilla: misc refactoring

### PR DESCRIPTION
**Pull request by booxter**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [X] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/bugzilla/default.nix [hunk 1]
```diff
@@ -3,26 +3,28 @@
   buildPythonPackage,
   fetchPypi,
   requests,
+  responses,
   pytestCheckHook,
   glibcLocalesUtf8,
 }:
 
 buildPythonPackage rec {
   pname = "bugzilla";
```
Comment (dotlambda) [inline pkgs/development/python-modules/bugzilla/default.nix:12]:
```suggestion
  pname = "python-bugzilla";
```

File: pkgs/development/python-modules/bugzilla/default.nix [hunk 2]
```diff
@@ -3,26 +3,28 @@
...
-  version = "3.2.0";
+  version = "3.3.0";
   format = "setuptools";
 
   src = fetchPypi {
-    pname = "python-${pname}";
+    pname = "python_${pname}";
     inherit version;
-    sha256 = "TvyM+il4N8nk6rIg4ZcXZxW9Ye4zzsLBsPJ5DweGA4c=";
+    sha256 = "sha256-4YIgFx4DPrO6YAxNE5NZ0BqhrOwdrrxDCJEORQdj3kc=";
```
Comment (dotlambda) [inline pkgs/development/python-modules/bugzilla/default.nix:19]:
```suggestion
    hash = "sha256-4YIgFx4DPrO6YAxNE5NZ0BqhrOwdrrxDCJEORQdj3kc=";
```

File: pkgs/development/python-modules/bugzilla/default.nix [hunk 3]
```diff
@@ -3,26 +3,28 @@
...
   };
 
   propagatedBuildInputs = [ requests ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/bugzilla/default.nix:22]:
```suggestion
  build-system = [ setuptools ];

  dependencies = [ requests ];
```

File: pkgs/development/python-modules/bugzilla/default.nix [hunk 4]
```diff
@@ -3,26 +3,28 @@
...
+  responses,
   pytestCheckHook,
   glibcLocalesUtf8,
 }:
 
 buildPythonPackage rec {
   pname = "bugzilla";
-  version = "3.2.0";
+  version = "3.3.0";
   format = "setuptools";
```
Comment (dotlambda) [inline pkgs/development/python-modules/bugzilla/default.nix:14]:
```suggestion
  pyproject = true;
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
The package should be called `python312Packages.python-bugzilla`.

**Comment by booxter** [comment]
@dotlambda That is not a new package. I am not sure I should rename it here, or touch the build system configuration. This PR only bumps the version. I assume there are good reasons to do the proposed refactoring, but perhaps these changes belong to a separate PR? Let me know, I am new to the project and don't know what is acceptable here.

**Comment by dotlambda** [comment]
You should make all changes but can split them into separate commits.

**Comment by booxter** [comment]
The version bump already happened in https://github.com/NixOS/nixpkgs/pull/349185 so now it's just refactoring PR...

**Comment by Frontear** [comment]
Out of curiosity, is there a reason we are changing this? I don't think it's a bad change, its just not a very useful change imo. Having to type an extra python- prefix to a package already part of a python3Packages namespace seems a little repetitive at best (I am aware other packages do this)

**Comment by booxter** [comment]
@Frontear this started as a version bump; @dotlambda asked to include these refactoring changes. Then a different PR merged with the version bump, so only the refactoring left to merge.

AFAIU the rename is to match the name of the package with the one in PyPI. But I will let @dotlambda to confirm. (I don't think the rest of changes are controversial.)

**Comment by dotlambda** [comment]
> AFAIU the rename is to match the name of the package with the one in PyPI.

Indeed.

**Review by dotlambda — APPROVED** [review]

**Pull request merged**